xlators/mgmt/glusterd/src/glusterd-geo-rep.c

```diff
@@ -4574,19 +4574,19 @@ glusterd_read_status_file(glusterd_volinfo_t *volinfo, char *secondary,
 
     primary = volinfo->volname;
 
+    len = snprintf(temp_conf_path, sizeof(temp_conf_path),
+                   "%s/" GSYNC_CONF_TEMPLATE, priv->workdir);
+    if ((len < 0) || (len >= sizeof(temp_conf_path))) {
+        return -1;
+    }
+
```

Comment:
Instead of moving to the top, we could use `goto out` below itself. And, in the `out` section, a check for `temp_in` could be added before GF_FREE

Reply:
The leak was for the `confd` dict and if we do `goto out`, we need to explicitly do some checks in the out section to identify where it failed. Because in all other places below this, if anything fails it returns 0. But in this section it returns -1. Just moving this to top will solve the leak without any fuss.

Reply:
I think we should return ret instead of returning 0, As we can see in the next statement if config file is not available it is throwing an ERROR statement but returning an 0 even operation is failed so the correct behavior should be return ret.